sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -119,8 +119,15 @@ case class GpuGetArrayItem(child: Expression, ordinal: Expression)
 
   override def doColumnar(lhs: GpuColumnVector, ordinal: Scalar): ColumnVector = {
     // Need to handle negative indexes...
-    if (ordinal.isValid && ordinal.getInt >= 0) {
-      lhs.getBase.extractListElement(ordinal.getInt)
+    if (ordinal.isValid) {
+      val minNumElements = lhs.getBase.countElements.min.getInt
+      if ( (ordinal.getInt < 0 | minNumElements < ordinal.getInt + 1) & failOnError) {
```

Comment:
why are we using bitwise or/and instead of logical ones?

Reply:
I must be drunk...Fix.